services/intersection-api/api/src/test/java/us/dot/its/jpo/ode/api/controllers/live/StompSessionControllerTest.java

```diff
@@ -0,0 +1,105 @@
+package us.dot.its.jpo.ode.api.controllers.live;
+
+import static org.mockito.Mockito.*;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.web.socket.messaging.AbstractSubProtocolEvent;
+import org.springframework.web.socket.messaging.SessionConnectEvent;
+import org.springframework.web.socket.messaging.SessionDisconnectEvent;
+import us.dot.its.jpo.ode.api.kafka.KafkaListenerControlService;
+
+import java.util.HashMap;
+import java.util.Map;
+
+class StompSessionControllerTest {
+
+    @Mock
+    KafkaListenerControlService listenerControlService;
+
+    StompSessionController controller;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+        controller = new StompSessionController(listenerControlService);
+    }
+
+    @Test
+    void testHandleSessionConnectEvent_StartListeners() {
+        // Mock event with sessionId header
+        Map<String, Object> headersMap = new HashMap<>();
+        headersMap.put("simpSessionId", "session1");
+        MessageHeaders headers = new MessageHeaders(headersMap);
+        Message<byte[]> message = mock(Message.class);
+        when(message.getHeaders()).thenReturn(headers);
+
+        AbstractSubProtocolEvent event = mock(SessionConnectEvent.class);
+        when(event.getMessage()).thenReturn(message);
+
+        controller.handleSessionConnectEvent((SessionConnectEvent) event);
+
+        // Should call startListeners since it's the first session
+        verify(listenerControlService).startListeners();
+    }
+
+    @Test
+    void testHandleSessionConnectEvent_NullSessionIdThrows() {
+        MessageHeaders headers = new MessageHeaders(new HashMap<>());
+        Message<byte[]> message = mock(Message.class);
+        when(message.getHeaders()).thenReturn(headers);
+
+        AbstractSubProtocolEvent event = mock(SessionConnectEvent.class);
+        when(event.getMessage()).thenReturn(message);
+
+        try {
```

Comment:
question: in many of our other tests we use the `assertThrows` rather than try/catch without an assert. Is there a reason for the swap?